lib/Motor/RoverArmMotor.cpp

```diff
@@ -0,0 +1,588 @@
+    }
+    else if (escType == BLUE_ROBOTICS)
+    {
+        // Max is actually 400 but this is safer.
+        internalPIDInstance = new PID(PID_DT, SERVO_MAX_OUTPUT, -SERVO_MAX_OUTPUT, regP, regI, regD);
+    }
+
+    /*------------------Get setpoint------------------*/
+    // Get current location and set it as setpoint. Essential to prevent jerkiness
+    // as the microcontroller initializes.
+    // adcResult = internalAveragerInstance.reading(analogRead(encoder));
+    // after setup, currentAngle is same as setpoint
+    delay(100 * DELAY_FACTOR);                       // safety
+    int error = get_current_angle_sw(&currentAngle); // fix setpoint not equal to current angle
+    if (error == -1)
+    {
+        encoder_error = 1;
+        return;
+    }
+    setpoint = currentAngle;
+    lastAngle = currentAngle;
+
+    /*------------------Set PID parameters------------------*/
+    regKp = regP;
+    regKi = regI;
+    regKd = regD;
+
+    // if(brake)  engage_brake(); //use brake if there is one
+    if (_brake_pin != -1)
+        engage_brake();        // use brake if there is one
+    delay(250 * DELAY_FACTOR); // wait for the motor to start up
+    this->stop();              // stop the motor
+
+    /*------------------Mastering------------------*/
+#if MASTERING == 1
+    delay(250 * DELAY_FACTOR); // wait for the motor to start up
+    this->reverse();
+    delay(250 * DELAY_FACTOR); // wait for the motor to start up
+    this->reverse();
+#endif
+
+    return;
+}
+
+int positive_rezeros = 0;
+double real_angle = 0;
+
+// Needs to be called in each loop
+void RoverArmMotor::tick()
+{
+    if (stop_tick)
+    {
+        this->stop(); // stop the motor
+    }
+    /*------------------Check limit pins------------------*/
+    // Print limit pins status
+    if (_limit_pin_max != -1 && _limit_pin_min != -1)
+    {
+        if (digitalRead(_limit_pin_max) == LOW)
+        {
+            this->stop();
+            this->reverse(10);
+            return;
+        }
+        if (digitalRead(_limit_pin_min) == LOW)
+        {
+            this->stop();
+            this->forward(10);
+            return;
+        }
+    }
+
+    /*------------------Get current angle------------------*/
+    int error = get_current_angle_sw(&currentAngle);
+    if (error == -1)
+    {
+        encoder_error = 1;
+        output = 0;
+        this->stop(); // stop the motor
+        return;
+    }
+    else
+    {
+        encoder_error = 0;
+    }
+
+    //------------------Remove jitter------------------//
+    // If the change in angle is less than the threshold, return early.
+    double diff;
+    if (wrist_waist)
+    {
+        diff = min(abs(currentAngle - setpoint), angle_full_turn - abs(currentAngle - setpoint));
+    }
+    else
+    {
+        diff = abs(currentAngle - setpoint);
+    }
+    if (diff < error_range)
+    {
+        output = 0;
+        this->engage_brake();
```

Comment:
Mathew and my suggestion would apply here for having the motor still on helping produce torque along side the break